File: 1006-Clumsy/1006.go
func clumsy(N int) (ans int) {
	stk := []int{N}
	N--

	index := 0
	for N > 0 {
		switch index % 4 {
		case 0:
			stk[len(stk)-1] *= N
		case 1:
			stk[len(stk)-1] /= N
		case 2:
			stk = append(stk, N)
		default:
			stk = append(stk, -N)
		}
		N--
		index++
	}

	for _, v := range stk {
		ans += v
	}
	return
}

File: 1006-Clumsy/1006.java
class Solution {
  public int clumsy(int N) {
    Deque<Integer> stack = new LinkedList<Integer>();
    stack.push(N);
    N--;

    int index = 0;
    while (N > 0) {
      if (index % 4 == 0) {
        stack.push(stack.pop() * N);
      } else if (index % 4 == 1) {
        stack.push(stack.pop() / N);
      } else if (index % 4 == 2) {
        stack.push(N);
      } else {
        sta.push(-N);
      }
      index++;
      N--;
    }
    int sum = 0;
    while (!stack.isEmpty()) {
      sum += stack.pop();
    }
    return sum;
  }
}